src/jit/codegencommon.cpp

```diff
@@ -729,12 +729,7 @@ void Compiler::compUpdateLifeVar(GenTreePtr tree, VARSET_TP* pLastUseVars)
     // verifying that we see the var nodes in execution order
     if (ForCodeGen)
     {
-        if (tree->gtOper == GT_OBJ)
-        {
-            // The tree must have the particular form OBJ(ADDR(LCL)); no need to do the check below.
-            assert(indirAddrLocal != NULL);
-        }
-        else if (tree->OperIsIndir())
+        if (tree->OperIsIndir())
         {
```

Comment:
I'm not sure that I fully grasp what this is/was trying to do (I note that the asserts below are #if'd out), but I'll attempt to reword it.

---
In reply to: [75345844](https://github.com/dotnet/coreclr/pull/6780#discussion_r75345844) [](ancestors = 75345844)